src/ComplexType.php

```diff
@@ -64,22 +68,31 @@ protected function generateClass()
         if ($this->class != null) {
             throw new Exception("The class has already been generated");
         }
+        $constructorParamsDefaultToNull = $this->config->get('constructorParamsDefaultToNull');
 
         $classBaseType = $this->getBaseTypeClass();
 
-        $this->class = new PhpClass(
-            $this->phpIdentifier,
-            false,
-            $classBaseType,
-            null,
-            false,
-            $this->abstract
-        );
+        $this->class = (new ZendClassGenerator())
+            ->setName($this->phpIdentifier)
+            ->setNamespaceName(
+                empty($this->config->get('namespaceName'))
+                    ? null
+                    : $this->config->get('namespaceName'))
+            ->setFlags(
+                $this->abstract
+                    ? ZendClassGenerator::FLAG_ABSTRACT
+                    : null)
+            ->setExtendedClass($classBaseType);
+
+        $constructor = (new MethodGenerator('__construct'));
+        $constructor
```

Comment:
Nitpicking: Is there any reason to add this line? I would think we could call `setFlags()` etc. fluidly.